### DISCUSSION TITLE
Keycloak Authentication

### DISCUSSION THREAD

**Discussion by M35a2** [horilla-opensource/horilla#428] [Ideas]
Has anyone worked on using [Keycloak](https://www.keycloak.org/) for authentication with Horilla? 

If not, I was going to give it a shot and submit a PR for it if it is successful.

**Comment by horilla-opensource** [comment]
> Has anyone worked on using [Keycloak](https://www.keycloak.org/) for authentication with Horilla?
> 
> If not, I was going to give it a shot and submit a PR for it if it is successful.

Hi @M35a2 ,

Thank you for the contributions. 

Currently no one is working on the keycloak authentication. Please feel free to do so.

With Regards,
**Team Horilla**

**Comment by AZAZAZZA** [comment]
could you please provide details about how to configure the secret server to store and retrieve the secrets from server . 
Please let me know if any functionality to configure secret server .

**Comment by horilla-opensource** [comment]
Hi @AZAZAZZA ,

Currently, Horilla relies on environment variables to manage secrets securely. 

Could you clarify what specific functionality you're looking for in terms of a secret server?

Best Regards,
**Team Horilla**

**Reply by M35a2** [reply]
I think @AZAZAZZA  means keycloak vault. 
in which case...
@AZAZAZZA , i dont use keycloak vault, i rely on hashicorp vault. So i wouldnt know - i would pose that question in the keycloak org discussions

**Reply by AZAZAZZA** [reply]
Thank you so much for the quick response.
Could you kindly provide detailed instructions on how to integrate Hashi Corp with Horilla?

**Comment by M35a2** [comment]
We are currently working on this in our development lab - i will update this discussion if/when we get it working. 
our solution right now relies on migrating to [django-allauth](https://docs.allauth.org/en/latest/) for authentication in horilla, then leveraging OAUTH/OPENID to authenticate the user's JWT.

**Comment by horilla-opensource** [comment]
**Hi @M35a2 and @AZAZAZZA**,

Thank you for your contributions and discussions on this. 

Currently, Horilla does not have built-in support for Keycloak Vault or HashiCorp Vault. 

However, we appreciate your efforts in exploring these integrations.

We’re happy to have you both as part of the Horilla community.

Best Regards,
**Team Horilla**
